sequence_processing_pipeline/NuQCJob.py

```diff
@@ -0,0 +1,368 @@
+                         f"OUTPUT={self.output}", f"TMPDIR={self.tmpdir}"]
+
+        job_params = ['-J', self.batch_prefix, '--array', batch_count,
+                      '--mem', '20G', '--export', ','.join(export_params)]
+
+        # job_script_path formerly known as:
+        #  process.multiprep.pangenome.adapter-filter.pe.sbatch
+        job_info = self.submit_job(job_script_path,
+                                   # job_parameters - was None
+                                   ' '.join(job_params),
+                                   # script_parameters
+                                   None,
+                                   # assume we want to exec from the log_path
+                                   # for now.
+                                   exec_from=self.log_path,
+                                   callback=callback)
+
+        job_id = job_info['job_id']
+        logging.debug(f'QCJob {job_id} completed')
+
+        '''
+        for project in self.project_data:
+            project_name = project['Sample_Project']
+            needs_human_filtering = project['HumanFiltering']
+
+            source_dir = join(self.output_path, project_name)
+
+
+            # TODO: UNCONFIRMED COMPLETED FILES WILL BE CREATED
+            # confirm Slurm job was successful by using .completed files.
+            # if not self._get_failed_indexes(project_name, job_id):
+            #     raise PipelineError("QCJob did not complete successfully.")
+
+            ### DANIEL'S NEW CODE DOES NOT APPEAR TO FILTER FOR
+                ZERO-LENGTH FILES ###
+
+            # determine where the filtered fastq files can be found and move
+            # the 'zero-length' files to another directory.
+            if needs_human_filtering is True:
+                filtered_directory = join(source_dir, 'filtered_sequences')
+            else:
+                filtered_directory = join(source_dir, 'trimmed_sequences')
+
+            empty_files_directory = join(source_dir, 'zero_files')
+            self._filter_empty_fastq_files(filtered_directory,
+                                           empty_files_directory,
+                                           self.minimum_bytes)
+        '''
+
+    def _process_sample_sheet(self):
+        sheet = KLSampleSheet(self.sample_sheet_path)
+        valid_sheet = validate_and_scrub_sample_sheet(sheet)
+
+        if not valid_sheet:
+            s = "Sample sheet %s is not valid." % self.sample_sheet_path
+            raise PipelineError(s)
+
+        header = valid_sheet.Header
+        chemistry = header['chemistry']
+
+        if header['Assay'] not in Pipeline.assay_types:
+            s = "Assay value '%s' is not recognized." % header['Assay']
+            raise PipelineError(s)
+
+        needs_adapter_trimming = (
+                    header['Assay'] in [Pipeline.METAGENOMIC_PTYPE,
+                                        Pipeline.METATRANSCRIPTOMIC_PTYPE])
```

Comment:
The PE filtering is not appropriate for metatranscriptomic data

Reply:
@antgonza @wasade This is an easy enough fix for me to make, but I'm suddenly wondering if adapter trimming previous metatranscriptomic runs had an adverse affect on their output. Currently we basically choose between three options - adapter-trimming (fastp by itself), adapter-trimming + human-filtering (fastp->minimap2->samtools chain), or neither (raw fastq files pass through without additional QC). Is this something to worry about?

Reply:
All prior adapter filtering with fastp, if it used paired end mode, did not actually filter adapters

Any existing human read removal on metatranscriptomic data using the minimap2 approach is bad

Reply:
See comment below.